JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts

```diff
@@ -513,6 +513,142 @@ class AppInsightsTests extends TestClass {
+                test(() => appInsights.trackPageView());
+                test(() => appInsights.trackMetric("testMetric", 0));
+                test(() => appInsights.trackException(new Error()));
+                test(() => appInsights.trackTrace("testTrace"));
+
+                // teardown
+                trackStub.restore();
+            }
+        });
+
+        this.testCase({
+            name: "AppInsightsTests: set authenticatedId and accountId context is applied",
+            test: () => {
+                // setup
+                var appInsights = new Microsoft.ApplicationInsights.AppInsights(this.getAppInsightsSnippet());
+
+                appInsights.setAuthenticatedUserContext("10001", "account33");
+
+                var trackStub = sinon.stub(appInsights.context._sender, "send");
+
+                // verify
+                var test = (action) => {
+                    action();
+                    this.clock.tick(1);
+                    var envelope = this.getFirstResult(action, trackStub);
+                    var contextKeys = new AI.ContextTagKeys();
+                    Assert.equal("10001", envelope.tags[contextKeys.userAuthUserId], "user.authenticatedId");
+                    Assert.equal("account33", envelope.tags[contextKeys.userAccountId], "user.accountId");
+
+                    trackStub.reset();
+                };
+
+                // act
+                test(() => appInsights.trackEvent("testEvent"));
+                test(() => appInsights.trackPageView());
+                test(() => appInsights.trackMetric("testMetric", 0));
+                test(() => appInsights.trackException(new Error()));
+                test(() => appInsights.trackTrace("testTrace"));
+
+                // teardown
+                trackStub.restore();
+            }
+        });
+
+        this.testCase({
+            name: "AppInsightsTests: set authenticatedId and accountId context is applied with non-ascii languages",
+            test: () => {
+                // setup
+                var appInsights = new Microsoft.ApplicationInsights.AppInsights(this.getAppInsightsSnippet());
+
+                appInsights.setAuthenticatedUserContext("myuser中国话", "اللغةالعربيةهيجميلةעבריתזהנחמד");
+
+                var trackStub = sinon.stub(appInsights.context._sender, "send");
+
+                // verify
+                var test = (action) => {
+                    action();
+                    this.clock.tick(1);
+                    var envelope = this.getFirstResult(action, trackStub);
+                    var contextKeys = new AI.ContextTagKeys();
+                    Assert.equal("myuser中国话", envelope.tags[contextKeys.userAuthUserId], "user.authenticatedId is correct, special characters removed");
+                    Assert.equal("اللغةالعربيةهيجميلةעבריתזהנחמד", envelope.tags[contextKeys.userAccountId], "user.accountIdis correct, special characters removed");
```

Comment:
that's hard to maintain, could you please switch to English alphabet or use special characters explicitly (e.g. with '\x' syntax).